app/channel.go

```diff
@@ -2622,10 +2704,32 @@ func (a *App) MarkChannelsAsViewed(channelIDs []string, userID string, currentSe
 	for _, channelID := range channelsToClearPushNotifications {
 		a.clearPushNotification(currentSessionId, userID, channelID)
 	}
+
+	if !collapsedThreadsSupported {
```

Comment:
For lines 14,15 in the unreads UX spreadsheet.

> When opening a channel from a client that doesn't support CRT or user has CRT turned off, both the channel and the threads in that channel are marked as read on desktop

May be out of scope or done in a follow up PR, but is this block where we should handle that?

Reply:
For the users that have CRT turned off?
Supporting CRT turned off will make some of the conditionals here a little more complicated. This PR is already too big, so I'd like to make those changes in a separate PR. 
We have a ticket for the issue:  https://mattermost.atlassian.net/browse/MM-35699

Reply:
> For the users that have CRT turned off?

Yes, and I agree should be done separately. More-or-less just a question about that. Sounds good on https://mattermost.atlassian.net/browse/MM-35699 👍